### PR TITLE
PART 3: Add UploadArchiveToGithubStorageWithMultiPart method

### PR DESCRIPTION
**Pull request by begonaguereca**
This pull request introduces a new feature for uploading archives to GitHub storage and includes corresponding unit tests. 

We are also updating the content types of Stream and Multipart uploads in the SendAsync method. 

- [x] Did you write/update appropriate tests
- [ ] Release notes updated (if appropriate)
- [ ] Appropriate logging output
- [ ] Issue linked
- [ ] Docs updated (or issue created)
- [ ] New package licenses are added to `ThirdPartyNotices.txt` (if applicable)

### CODE REVIEW

File: src/Octoshift/Services/GithubApi.cs [hunk 1]
```diff
@@ -1047,6 +1048,33 @@ mutation abortRepositoryMigration(
...
+        if (isMultipart)
+        {
+            var url = $"https://uploads.github.com/organizations/{org.EscapeDataString()}/gei/archive/blobs/uploads";
+
+            using var content = new MultipartFormDataContent
+            {
+                { httpContent, "archive", archiveName }
+            };
+
+            response = await _client.PostAsync(url, content);
```
Comment (begonaguereca) [inline src/Octoshift/Services/GithubApi.cs:1065]: @ArinGhazarian is this what you envisioned for the return type? https://github.com/github/gh-gei/pull/1275

File: src/Octoshift/Services/GithubApi.cs [hunk 2]
```diff
@@ -1047,6 +1048,33 @@ mutation abortRepositoryMigration(
...
+        }
+        else
+        {
+            var url = $"https://uploads.github.com/organizations/{org.EscapeDataString()}/gei/archive\\?name\\={archiveName}";
```
[the remark this answers is not in the document]
Comment (begonaguereca) [reply src/Octoshift/Services/GithubApi.cs]: Fixed!

File: src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs [hunk 3]
```diff
@@ -3462,6 +3463,62 @@ await _githubApi.Invoking(api => api.AbortMigration(migrationId))
             .WithMessage(expectedErrorMessage);
     }
 
+    [Fact]
+    public async Task UploadArchiveToGithubStorage()
```
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs:3467]:
```suggestion
    public async Task UploadArchiveToGithubStorage_Should_Upload_Stream_Content()
```

File: src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs [hunk 4]
```diff
@@ -3462,6 +3463,62 @@ await _githubApi.Invoking(api => api.AbortMigration(migrationId))
...
+        const bool isMultipart = false;
+        const string archiveName = "archiveName";
+
+        // Using a MemoryStream as a valid stream implementation
+        using var archiveContent = new MemoryStream(new byte[] { 1, 2, 3 });
+        var expectedArchiveId = "123456";
+        var jsonResponse = $"{{ \"archiveId\": \"{expectedArchiveId}\" }}";
+
+        _githubClientMock
+            .Setup(m => m.PostAsync(It.IsAny<string>(), It.IsAny<HttpContent>(), null))
```
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs:3480]: Let's make sure that `PostAsync` receives a `StreamContent`:

```suggestion
            .Setup(m => m.PostAsync(It.IsAny<string>(), It.IsAny<StreamContent>(), null))
```

File: src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs [hunk 5]
```diff
@@ -3462,6 +3463,62 @@ await _githubApi.Invoking(api => api.AbortMigration(migrationId))
...
+        var actualStringResponse = await _githubApi.UploadArchiveToGithubStorage(org, isMultipart, archiveName, archiveContent);
+
+        // Assert
+        expectedStringResponse.Should().Be(actualStringResponse);
+
+    }
+
+
+    [Fact]
+    public async Task UploadArchiveToGithubStorageWithMultiPart()
```
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs:3495]: We use snaked cased test names throughout the project:

```suggestion
    public async Task UploadArchiveToGithubStorage_Should_Upload_Multipart_Content()
```

File: src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs [hunk 6]
```diff
@@ -3462,6 +3463,62 @@ await _githubApi.Invoking(api => api.AbortMigration(migrationId))
...
+        const string archiveName = "archiveName";
+
+        // Using a MemoryStream as a valid stream implementation
+        using var archiveContent = new MemoryStream(new byte[] { 1, 2, 3 });
+
+        var expectedArchiveId = "123456";
+        var jsonResponse = $"{{ \"archiveId\": \"{expectedArchiveId}\" }}";
+
+        _githubClientMock
+            .Setup(m => m.PostAsync(It.IsAny<string>(), It.IsAny<HttpContent>(), null))
```
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs:3509]:
```suggestion
            .Setup(m => m.PostAsync(It.IsAny<string>(), It.IsAny<MultipartFormDataContent>(), null))
```

**Review by ArinGhazarian — CHANGES_REQUESTED** [review]
Dropped some comments mostly around simplifying the code and unit tests. Tests need to be added for `GithubClient` too

**Pull request merged**